open-apparel-registry/open-apparel-registry
# Integrate Mailchimp

## Overview

The OAR registration form includes an option to sign up for emails. Users who check that box should automatically be added to the OAR Mailchimp account.

Connects [#27](https://github.com/open-apparel-registry/open-apparel-registry-clients/issues/27)

## Demo

<img width="987" alt="Screen Shot 2021-07-01 at 9 33 12 AM" src="https://user-images.githubusercontent.com/21046714/124137139-65345000-da53-11eb-9c69-496310106cc1.png">

## Notes

Once we have merged this, we will need to add the production API key and List ID to Terraform officially. We also need to set up a free-tier OAR Mailchimp account to provide us with a staging API key and List ID, which will also be added to Terraform. 

## Testing Instructions

* Create a free-tier Mailchimp account and configure (OR contact me for my list ID and API key):
     - API Key:  From the left main menu, click on your Username, then select "Account" in the menu. From the account page select "Extras", "API Keys". Your API Key will be listed in the table labeled "Your API Keys".
     - List ID/Audience ID:  When viewing your MailChimp Lists, select the list you wish to offer user sign-up, then click "Settings" and then "List/Audience name & defaults". Your list id will be at the top of the right column below List ID or Audience ID.
     - Merge Tags:  When viewing your MailChimp Lists, select the list you wish to offer user sign-up, then click "Settings" and then “Audience Fields and Merge Tags”. Match the configuration to: 
![Screen Shot 2021-06-30 at 5 19 15 PM](https://user-images.githubusercontent.com/21046714/124137853-16d38100-da54-11eb-97c8-48e7814e05a7.png)
* In your .env file, add these two lines, with your API key and List ID:
```
MAILCHIMP_API_KEY=
MAILCHIMP_LIST_ID=
```
* Run `./scripts/server` and navigate to [registration](http://localhost:6543/auth/register)
* Register with the 'subscribe' checkbox selected. Note that obviously 'fake' emails such as '123@example.com' will be rejected by Mailchimp. 
* View your contacts in Mailchimp. You should see your new user has been added to the list, and includes the email, organization name, and contributor type.

## Checklist
